index.js

```diff
@@ -72,6 +72,23 @@ module.exports = {
     "simple-import-sort/imports": "error",
     "simple-import-sort/exports": "error",
     "sort-destructure-keys/sort-destructure-keys": "error",
+    /**
+     * List of sonarjs rules we turned off after release of eslint-plugin-sonarjs v2
+     * Some rules are too restrictive for common use cases,
+     * and others are already covered by other plugins used here.
+     */
+    "sonarjs/different-types-comparison": "off",
+    "sonarjs/function-return-type": "off",
+    "sonarjs/no-clear-text-protocols": "warn",
+    "sonarjs/no-ignored-exceptions": "off",
+    "sonarjs/no-misused-promises": "off",
+    "sonarjs/no-nested-functions": "off",
```

Comment:
It looks like this rule only reports on the arrow callback, rather than highlighting all the associated lines of logic. Seems worth considering as a `warn` for very nested logic.

Here's an example of one that is probably worth `// eslint-disable-next-line`-ing:

<img width="1131" alt="Screenshot 2024-09-12 at 12 59 12 PM" src="https://github.com/user-attachments/assets/a2330f89-176a-4fab-9354-2d578634f16c">

Here is an example of many `.maps` within JSX return that would likely resolve the rule through extracting that logic into a separate component:

<img width="1053" alt="Screenshot 2024-09-12 at 1 02 49 PM" src="https://github.com/user-attachments/assets/e4825e3b-ce92-42d5-9f3e-8548ab49c35b">

(We could also push the default threshold from 4 to 5 to catch only the most-nested cases: `"sonarjs/no-nested-functions": ["warn", { threshold: 5 }]`).